recipes/botan/all/conanfile.py

```diff
@@ -377,9 +421,15 @@ def _configure_cmd(self):
 
         build_flags.append('--without-pkg-config')
 
+        if self.settings.os == 'Linux':
+            # The current Ubuntu 18.04 runners in Conan Center Index don't have getrandom/getentropy,
+            # which is enabled by default starting in Botan 3.0.0
+            build_flags.append('--without-os-feature=getrandom')
+            build_flags.append('--without-os-feature=getentropy')
```

Comment:
Maybe this should be removed again, as the CI is currently not not run for 3.0.0 (there this is enabled by default) because of old compilers anyway?

Reply:
Not compilers, but glibc version. This error is listed in the upstream: https://github.com/randombit/botan/issues/2607#issuecomment-1382826977 but we can't do nothing about because Docker images used in CCI are not running on latest Ubuntu version to avoid breaking users with non-updated distros. I would suggest adding an option like `with_getrandom: False` where it uses `False` by default.

In CCI we try to preserve the upstream behavior always, but at same time, packages should be built, so this case deserves an exception.

Thank you for checking it !

Reply:
@uilianries is on the case with this one :)

Reply:
I just created a PR to Botan: https://github.com/randombit/botan/pull/3685

The `getentropy` is indeed a new feature, not available as method, but also syscall.

I did some tests using the docker image `conanio/gcc5` with the follow code:

```c
#include <stdlib.h>
#include <stdio.h>
#include <unistd.h>

#include <sys/syscall.h>
#include <linux/random.h>


int main(int arg, char *argv[]) {
    char buffer[5];
    size_t length = 5;
    int result_getrandom = 0;
    int result_getentropy = 0;

#if defined(USE_SYSCALL)
    result_getrandom = syscall(SYS_getrandom, buffer, length, 0);
    result_getentropy = syscall(SYS_getentropy, buffer, length);
#else
    result_getrandom = getrandom(buffer, length, 0);
    result_getentropy = getentropy(buffer, length);
#endif

    if (result_getrandom == -1) {
        perror("getrandom");
        return EXIT_FAILURE;
    }
    if (result_getentropy == -1) {
        perror("getrandom");
        return EXIT_FAILURE;
    }

    return EXIT_SUCCESS;
}
```

When using syscall for getrandom, it works. The getentropy needs a newer glibc version.

Reply:
@RubenRBS @lieser Good news, my PR was merged into Botan, so you can apply it to the package and enable getrandom always: https://github.com/randombit/botan/pull/3685

Reply:
We probably want to remove the `build_flags.append('--without-os-feature=getrandom')` line (but not the one for `getentropy`), as we have patches to fix it for `getrandom` now.